torch/distributed/_sharded_tensor/api.py

```diff
@@ -136,7 +136,7 @@ def __init__(
         if memory_format != torch.contiguous_format:
             raise ValueError('Only torch.contiguous_format memory_format is currently supported')
 
-        dims = list(size)
+        dims = list(*size) if len(size) == 1 else list(size)
```

Comment:
why `len(size) == 1` we still unpack the `size`? it seems to me when length==1, size will just become a tuple like `(3,)`, shouldn't this still work?

Reply:
The problem is when size is a list, what list(size) does is it creates a List[List[]]